vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java

```diff
@@ -418,8 +419,21 @@ private void doDisableOnClick() {
         }
     }
 
+
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        disableListener.remove();
+    }
+
     @Override
     protected void onAttach(AttachEvent attachEvent) {
+        if (disableListener == null) {
+            disableListener = addClickListener(buttonClickEvent -> {
```

Comment:
This could be problematic; because now (after re-attach) the disableOnClick listener runs after the other listener and therefor does not stop the click propagation anymore.. is there a way to register the clicklistener as "first" again?

Reply:
Yeah that is the challenging part here. I also wonder why this specific listener creates a leak in the first place. We have lots of components creating listeners on themselves. So we either have dozens of memory leaks, or there is some specific issue with this one. One noticable difference is that this component stores the registration as a field which other components do not do.

Reply:
Based on comment by @Artur- in the ticket, this PR might not be the proper way to fix the problem and actually provides only a partial help.